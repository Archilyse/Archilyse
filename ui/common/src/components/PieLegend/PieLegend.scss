@import '../../theme.scss';

.pie-legend {
  ul {
    margin-left: auto;
    padding-left: 0;
    list-style: none;

    li {
      color: $secondary-color;
      margin-bottom: $unit/2;
      display: flex;
      font-size: small;
      justify-content: flex-end;
      align-items: center;
      .bar {
        width: $unit * 0.44;
        height: $unit * 2.32;
        margin-left: $unit;
      }
    }
  }
}
