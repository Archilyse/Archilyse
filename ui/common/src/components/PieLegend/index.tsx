import React from 'react';
import { capitalize } from '../../modules';
import './PieLegend.scss';

const formatLegend = status => capitalize(status.toLowerCase());

export default ({ data, format = true, className = '', itemStyle = (item: string) => ({}) }) => {
  return (
    <div className={`pie-legend ${className}`}>
      <ul>
        {data.map(item => {
          return (
            <li key={item}>
              <p>{format ? formatLegend(item) : item}</p>
              <div className="bar" style={itemStyle(item)}></div>
            </li>
          );
        })}
      </ul>
    </div>
  );
};
