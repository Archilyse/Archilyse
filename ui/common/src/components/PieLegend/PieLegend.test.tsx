import * as React from 'react';
import { cleanup, render } from '@testing-library/react';
import PieLegend from './index';

afterEach(cleanup);

const MOCK_DATA = ['HNF', 'NNF', 'ANF', 'FF', 'VF'];

const MOCK_ANALYSIS_COLORS = {
  HNF: '#00B34A',
  ANF: '#FF3C2E',
  FF: '#FFB3AD',
  NNF: '#2EFF8C',
  VF: '#9E190C',
};

const legendItemStyle = item => ({
  backgroundColor: `${MOCK_ANALYSIS_COLORS[item]}`,
});

it('renders correctly', () => {
  const { container } = render(<PieLegend data={MOCK_DATA} itemStyle={legendItemStyle} />);
  expect(container).toMatchSnapshot();
});
