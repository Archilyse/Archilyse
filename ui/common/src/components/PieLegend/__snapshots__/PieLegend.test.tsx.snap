// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="pie-legend "
  >
    <ul>
      <li>
        <p>
          Hnf
        </p>
        <div
          class="bar"
          style="background-color: rgb(0, 179, 74);"
        />
      </li>
      <li>
        <p>
          Nnf
        </p>
        <div
          class="bar"
          style="background-color: rgb(46, 255, 140);"
        />
      </li>
      <li>
        <p>
          Anf
        </p>
        <div
          class="bar"
          style="background-color: rgb(255, 60, 46);"
        />
      </li>
      <li>
        <p>
          Ff
        </p>
        <div
          class="bar"
          style="background-color: rgb(255, 179, 173);"
        />
      </li>
      <li>
        <p>
          Vf
        </p>
        <div
          class="bar"
          style="background-color: rgb(158, 25, 12);"
        />
      </li>
    </ul>
  </div>
</div>
`;
